src/game/objects/traps/movable_block.c

```diff
@@ -243,9 +245,9 @@ void MovableBlock_Control(int16_t item_num)
 
     int16_t room_num = item->room_number;
     FLOOR_INFO *floor =
-        Room_GetFloor(item->pos.x, item->pos.y, item->pos.z, &room_num);
+        Room_GetFloor(item->pos.x, item->pos.y - 128, item->pos.z, &room_num);
```

Comment:
Am I right in thinking that `-128` is used to ensure we are getting the actual sector where the block is positioned, and so by moving up a little, we avoid `Room_GetFloor` traversing portals and hence returning sectors further down the stack? Might be an idea to change this to `-STEP_L / 2` for clarity.

Reply:
Ya good call. Fixed.